enclave-runtime/src/lib.rs

```diff
@@ -622,25 +632,33 @@ where
 	let mut validator = LightClientSeal::<PB>::unseal()?;
 	let mut nonce = NONCE.write().expect("Encountered poisoned NONCE lock");
 
-	sync_blocks_on_light_client(blocks_to_sync, &mut validator, &OcallApi, &mut *nonce)?;
+	sync_blocks_on_light_client(
+		blocks_to_sync,
+		&mut validator,
+		&OcallApi,
+		&GlobalFileStateHandler,
+		&mut *nonce,
+	)?;
 
 	// store updated state in light client in case we fail afterwards.
 	LightClientSeal::seal(validator)?;
 
 	Ok(())
 }
 
-fn sync_blocks_on_light_client<PB, V, O>(
+fn sync_blocks_on_light_client<PB, V, O, StateHandler>(
```

Comment:
Here you're using StateHandler, one function below only "S". Is that on purpose?